src/Form/Control/ScopeBuilder.php

```diff
@@ -261,8 +262,51 @@ class ScopeBuilder extends Control
                 self::OPERATOR_EMPTY,
                 self::OPERATOR_NOT_EMPTY,
             ],
+            'props' => [
+                'dateFormat' => 'Y-m-d',
+            ],
+        ],
+        'datetime' => [
+            'type' => 'custom-component',
+            'component' => 'DatePicker',
+            'inputType' => 'datetime',
+            'operators' => [
+                self::OPERATOR_TIME_EQUALS,
+                self::OPERATOR_TIME_DOESNOT_EQUAL,
+                self::OPERATOR_TIME_GREATER,
+                self::OPERATOR_TIME_GREATER_EQUAL,
+                self::OPERATOR_TIME_LESS,
+                self::OPERATOR_TIME_LESS_EQUAL,
+                self::OPERATOR_EMPTY,
+                self::OPERATOR_NOT_EMPTY,
+            ],
+            'props' => [
+                'dateFormat' => 'Y-m-d H:i',
+                'enableTime' => true,
+                'time_24hr' => true,
+            ],
+        ],
+        'time' => [
+            'type' => 'custom-component',
+            'component' => 'DatePicker',
+            'inputType' => 'time',
+            'operators' => [
+                self::OPERATOR_TIME_EQUALS,
+                self::OPERATOR_TIME_DOESNOT_EQUAL,
+                self::OPERATOR_TIME_GREATER,
+                self::OPERATOR_TIME_GREATER_EQUAL,
+                self::OPERATOR_TIME_LESS,
+                self::OPERATOR_TIME_LESS_EQUAL,
+                self::OPERATOR_EMPTY,
+                self::OPERATOR_NOT_EMPTY,
+            ],
+            'props' => [
+                'dateFormat' => 'H:i',
+                'enableTime' => true,
+                'noCalendar' => true,
+                'time_24hr' => true,
+            ],
```

Comment:
@georgehristov - Can you take a look at how datetime and time rules are defined. Time and Datetime can now by supported via vue-flatpickr .
Question: 
Do we need datetime at all?
Flatpickr vue component needs to have proper set of properties define to switch from date, time or datetime picker. I add them via $rule['props'] property. Is that ok? Do we need to switch format based on ui persistence? if yes, then where do we set the property. If not, should we just set them in javascript not in php?

Reply:
I remove props and now use rule['options'] see new comments.